stockfish/models.py

```diff
@@ -728,6 +719,208 @@ def will_move_be_a_capture(self, move_value: str) -> Capture:
+                src_file = groups[1]
+
+            # resolve source rank
+            src_rank = None
+            if len(groups[2]) == 1:
+                src_rank = groups[2]
+
+            # resolve capture
+            isCapture = groups[3] == "x"
+
+            # pawn conversion
+            turnsInto = groups[5].lstrip("=")
+
+            # resolve dst
+            dst = groups[4]
+
+            # resolve src
+            src = None
+            # find src
+            if src_file is not None and src_rank is not None:
+                src = f"{src_file}{src_rank}"
+            else:
+                possibleSrc = []
+                # run through all the squares and check all the pieces if they can move to the square
+                for file_as_unicode_int in range(ord("a"), ord("h") + 1):
+                    file = chr(file_as_unicode_int)
+                    if src_file is not None and src_file != file:
+                        continue
+                    for rank_as_int in range(1, 8 + 1):
+                        rank = str(rank_as_int)
+                        if src_rank is not None and src_rank != rank:
+                            continue
+                        src = f"{file}{rank}"
+                        if piece == self.get_what_is_on_square(
+                            src
+                        ) and self.is_move_correct(f"{src}{dst}{turnsInto}"):
+                            possibleSrc.append(src)
+                if len(possibleSrc) == 1:
+                    src = possibleSrc[0]
+                elif len(possibleSrc) == 0:
+                    pieceDesc = str(piece).replace("Piece.", "")
+                    if src_rank is not None and src_file is None:
+                        pieceDesc = pieceDesc + f" from rank {src_rank}"
+                    elif src_rank is None and src_file is not None:
+                        pieceDesc = pieceDesc + f" from file {src_file}"
+                    # no need to check for both since that is already covered above
+                    # no need to check for neither since no additional description is needed
+                    raise ValueError(f"No {pieceDesc} can go to {dst}")
+                else:
+                    pieceDesc = str(piece).replace("Piece.", "")
+                    raise ValueError(
+                        f"Could not determine which {pieceDesc} you want to move to {dst}"
+                    )
+            # build stockfish move
+            move = f"{src}{dst}{turnsInto}"
+            # check if resolved move is indeed a capture
+            if self.is_move_correct(move):
+                if (
+                    isCapture
+                    and turnsInto != ""
+                    and self.get_what_is_on_square(dst) is None
+                ) or (
+                    isCapture
+                    and turnsInto == ""
+                    and self.will_move_be_a_capture(move)
+                    == Stockfish.Capture.NO_CAPTURE
+                ):
```

Comment:
Since bug #99 was fixed, no need for this convoluted check. Use this instead.
```python
if (isCapture and self.will_move_be_a_capture(move) == Stockfish.Capture.NO_CAPTURE):
```